script/git-clone

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+#  Copyright (c) 2018, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+#    Description:
+#      A tool to download projects with the ability to apply dependent PRs.
+#
+
+set -euxo pipefail
+
+do_check()
+{
+    local project_name
+
+    project_name=$(git remote get-url origin | grep -oE '[^/:]+/[^/:]+\.git$' | cut -d. -f1)
+    echo "project name: ${project_name}"
+    depends_on_pr="$(grep -E "^Depends-On: *${project_name}" | cut -d# -f2)" || {
+        echo "No dependency on ${project_name} found."
+        return 0
+    }
+    echo "pr: #${depends_on_pr}"
+    git fetch --depth 1 origin "pull/${depends_on_pr}/head"
+    git checkout FETCH_HEAD
+}
+
+do_get_pr_body()
+{
+    local pr
+
+    pr="$(echo "${GITHUB_REF}" | cut -d/ -f3)"
+
+    # do not print GITHUB_TOKEN
+    set +x
+    curl -H "authorization: Bearer ${GITHUB_TOKEN}" -H 'content-type: application/json' \
+        "https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${pr}" \
+        | jq -r .body
+    set -x
+}
+
+try_git_clone()
+{
+    local dest_dir
+
+    dest_dir="$(git clone "$@" 2>&1 | tee | cut -d\' -f2)"
+
+    [[ ${GITHUB_ACTIONS+x} && ${GITHUB_REF-} =~ ^refs/pull/[0-9]+/merge ]] || {
+        echo 'Not a pull request.'
```

Comment:
We don't need this in master (merged) ?

Reply:
The goal of this feature is to allow a project's CI apply dependency PRs, so the process would be merge a PR and its dependencies all together.

Reply:
We are going to fail when merged to master. 

Meanwhile the current way of using a specific commit ID, even though a little bit cumbersome, works for both PR and branch, without any check fails.

Reply:
I don't think it a good idea to use a commit id which doesn't exist in master branch and having to change the workflow file from time to time. For PRs in OpenThread, it should be merged together with its dependency PRs, so I don't think the CI would fail in OpenThread project. The dependency project will fail only if it tracks OpenThread with submodule in its CI system, which I think we can change to use master branch to prevent such failures. Thoughts?

Reply:
Agree that it's a better solution than using fixed commit IDs. 

Again, we should checkout master branch when merged to master, right?